### PR TITLE
#11384 - Filler 2.0 - Svg does not draw clipPath definitions

### PR DESCRIPTION
**Pull request by Zeljko-Predjeskovic**

### CODE REVIEW

File: Svg/Basic Shapes/SvgImage.cs [hunk 1]
```diff
@@ -260,7 +261,7 @@ protected override void Render(ISvgRenderer renderer)
                         renderer.PopBoundable();
                     }
 
-                    if (this.ClipPath != null || !string.IsNullOrEmpty(this.Clip))
+                    if(Parent is not SvgUse)
```
Comment (gentledepp) [inline Svg/Basic Shapes/SvgImage.cs:264]: aha! so with a svguse, we must keep the clip?

File: Svg/Clipping and Masking/SvgClipPath.cs [hunk 2]
```diff
@@ -54,7 +53,7 @@ public GraphicsPath GetClipRegionPath(SvgVisualElement owner)
                 this._pathDirty = false;
             }
 
-            var result = cachedClipPath;
+            var result = cachedClipPath.Clone();
```
Comment (gentledepp) [inline Svg/Clipping and Masking/SvgClipPath.cs:56]: so one issue was that the same clippath-instance was transformed over and over again?

**Review by gentledepp — APPROVED** [review]
looks good. Please answer my two questions though so I understand what is going on :-)

File: Svg/Clipping and Masking/SvgClipPath.cs (hunk 2 above)
Comment (Zeljko-Predjeskovic) [reply Svg/Clipping and Masking/SvgClipPath.cs:56]: yes thats why we got an arithmetic overflow.

File: Svg/Basic Shapes/SvgImage.cs (hunk 1 above)
Comment (Zeljko-Predjeskovic) [reply Svg/Basic Shapes/SvgImage.cs:264]: Usually svg use has the image as an attribute and not child. Thats  why it should not clip again.

**Pull request merged**